services/audio/src/main/java/ca/bc/gov/tno/services/audio/config/AudioConfig.java

```diff
@@ -58,22 +64,28 @@ public AudioConfig() {
    * Creates a new instance of a AudioConfig object, initializes with
    * specified parameters.
    * 
-   * @param dataSource
+   * @param dataSource The data source for this configuration
+   * @param api The api based data service
    */
-  public AudioConfig(final DataSource dataSource) {
+  public AudioConfig(final DataSource dataSource, final TnoApi api) {
     super(dataSource);
 
     setAudioService((String) dataSource.getCode());
 
-    if (dataSource.getParent() != null) {
-      var parent = dataSource.getParent();
-      var connection = parent.getConnection();
-      setCaptureService(parent.getCode());
-      setStreamStartTime((long) connection.get("streamStart"));
-      setCaptureDir((String) connection.get("captureDir"));
-      setClipDir((String) connection.get("clipDir"));
-      setClipCmd((String) connection.get("clipCmd"));
-      setTimezone((String) connection.get("timeZone"));
+    try {
+      if (dataSource.getParentId() != null) {
+        var thisDs = api.getDataSource(dataSource.getCode());
+        var parent = thisDs.getParent();
+        var connection = parent.getConnection();
+        setCaptureService(parent.getCode());
+        setStreamStartTime((long) connection.get("streamStart"));
+        setCaptureDir((String) connection.get("captureDir"));
+        setClipDir((String) connection.get("clipDir"));
+        setClipCmd((String) connection.get("clipCmd"));
+        setTimezone((String) connection.get("timeZone"));
+      }
+    } catch(Exception e) {
+      logger.error("Exception initializing audio configuration.", e);
```

Comment:
Regrettably this will not register as a failure for the data source, as such it'll keep doing it over and over.

Reply:
The only reason I added that catch block was because it was throwing an exception where it wasn't throwing one before. This was really a stop-gap solution until I could run a permanent solution by you. If I change the constructor of DataSourceConfig, and this constructor, to throw ApiException I think that will address it. Should I go ahead and do that?

Reply:
If the exception doesn't bubble up to the event handler then it'll never get captured and recorded as such.  I'd remove they try+catch if you can and have the error handled by whatever is calling it.

Reply:
That was the first thing I tried, but the only way to code it was either catch and kill the exception or change the constructor of DataSourceConfig so that it also threw the ApiException. I didn't want to have to change the library if it could be avoided, but will do that so that it bubbles up all the way.

Reply:
I figured when I first saw this that it might be more complicated than it looks...and the reason is "Java"! In initConfigs() (which throws an api exception) you fill the sourceConfigs collection using the following lambda expression:

    approvedDataSources.forEach(ds -> sourceConfigs.getSources().add(new CaptureConfig(ds)));

The problem is that the constructor for CaptureConfig(ds) now throws an ApiException and because it's called inside the lambda expression it's not in the same scope as the rest of the method. As a result, the compiler requires the call to CaptureConfig(ds) to be wrapped in a try/catch block, which ruins the concise syntax and would consume the exception preventing it from bubbling up.

My suggestion is to use the following which will perform the same function and throw any ApiExceptions properly:

    for (var ds : approvedDataSources) {
      sourceConfigs.getSources().add(new CaptureConfig(ds));
    }

What do you think?

Reply:
I'm relying on your knowledge of java here.  If you think it'll work and is clean.  Go ahead.